proto/spire/api/types/federationrelationship.proto

```diff
@@ -20,17 +20,25 @@ message FederationRelationship {
         // Use SPIFFE Authentication endpoint profile.
         HTTPSSPIFFEProfile https_spiffe = 4;
     }
+
+    // Optional. The bundle for the trust domain. This field can be used to
+    // create/replace the bundle when the relationship is created or updated.
```

Comment:
Since the field will be in the `FederationRelationship` message and will not be specific for the `https_spiffe` profile, I’m wondering if it’s clear that for create/update the field is only useful when using the `https_spiffe` profile. If I’m creating a relationship with the https_web profile I may think that I need to fill it.

So we may say something like
_This field can be used to create/replace the bundle when the relationship is created or updated using the https_spiffe profile_

What do you think?

Reply:
Due to it's generic nature, I'd prefer not to have special case behavior for this field based on the profile of choice. In my opinion, if supplied, no matter the profile, it should be create/update the bundle for the trust domain.

Reply:
> Since the field will be in the FederationRelationship message and will not be specific for the https_spiffe profile, I’m wondering if it’s clear that for create/update the field is only useful when using the https_spiffe profile.

...

> In my opinion, if supplied, no matter the profile, it should be create/update the bundle for the trust domain.
> In other words, this field is simply a convenience for setting the trust domain bundle for the relationship at the same time the relationship is created, instead of needing to use two calls.

I think there is some use (call it a small convenience) even for other profiles as it should allow your workloads to begin federating immediately vs waiting for the first bundle endpoint conversation to take place (which may even fail).

I think we also want this bundle to be reflected in API responses regardless of the profile details?